planner/core/rule_predicate_push_down.go

```diff
@@ -173,6 +173,12 @@ func (p *LogicalJoin) PredicatePushDown(predicates []expression.Expression) (ret
 		p.OtherConditions = otherCond
 		leftCond = leftPushCond
 		rightCond = rightPushCond
+		if p.JoinType == AntiSemiJoin {
```

Comment:
I think this might be too strict. Look at the explain-test(tpc-h q21)this PR has affected. 
```sql
select
	s_name,
	count(*) as numwait
from
	supplier,
	lineitem l1,
	orders,
	nation
where
	s_suppkey = l1.l_suppkey
	and o_orderkey = l1.l_orderkey
	and o_orderstatus = 'F'
	and l1.l_receiptdate > l1.l_commitdate
	and exists (
		select
			*
		from
			lineitem l2
		where
			l2.l_orderkey = l1.l_orderkey
			and l2.l_suppkey <> l1.l_suppkey
	)
	and not exists (
		select
			*
		from
			lineitem l3
		where
			l3.l_orderkey = l1.l_orderkey
			and l3.l_suppkey <> l1.l_suppkey
			and l3.l_receiptdate > l3.l_commitdate
	)
	and s_nationkey = n_nationkey
	and n_name = 'EGYPT'
group by
	s_name
```
After this change, the subquery condition `l3.l_receiptdate > l3.l_commitdate` cannot be pushed down. This can casue a performance regression. 

Actually, although `not exists` and `not in` both use `anti-semi-join`, but they are not totally equal. I think we'd better fix this issue for `not in`, and not affect the behavior of `not exists`. 

Here are some reference you can take a look:
This is how mysql deals with `exists strategy`.
https://dev.mysql.com/doc/refman/8.0/en/subquery-optimization-with-exists.html

And this is how SparkSQL uses a `null aware anti join` for `not in`(I prefer this method to rewrite the `not in` condition). https://databricks-prod-cloudfront.cloud.databricks.com/public/4027ec902e239c93eaaa8714f173bcfc/2728434780191932/1483312212640900/6987336228780374/latest.html

Reply:
Here is how `null aware anti join` works (quoted from Apache Spark):
```
Expand the NOT IN expression with the NULL-aware semantic
to its full form. That is from:
  (a1,a2,...) = (b1,b2,...)
to
  (a1=b1 OR isnull(a1=b1)) AND (a2=b2 OR isnull(a2=b2)) AND ...
```

Reply:
Thanks，maybe I need to rehink about it. Please remove the review request.This commit is too hasty.

Reply:
You can check the `InOperand` field of `RightConditions`'s arguments to decide if we should disable pushdown for it. `InOperand` is introduced to differentiate the `AntiSemiJoin` / `LeftAntiSemiJoin` / `AntiLeftOuterSemiJoin` generated by `NOT IN` and `NOT EXISTS`.